geth/jail/cell_test.go

```diff
@@ -7,254 +7,92 @@ import (
-		w.Write([]byte(body)) //nolint: errcheck
-	}))
-	defer server.Close()
-
-	require := s.Require()
-
-	cell, err := s.jail.NewCell(testChatID)
-	require.NoError(err)
-	require.NotNil(cell)
-	defer cell.Stop()
-
-	dataCh := make(chan otto.Value, 1)
-	errCh := make(chan otto.Value, 1)
-
-	err = cell.Set("__captureSuccess", func(res otto.Value) { dataCh <- res })
-	require.NoError(err)
-	err = cell.Set("__captureError", func(res otto.Value) { errCh <- res })
-	require.NoError(err)
-
-	// run JS code for fetching valid URL
-	_, err = cell.Run(`fetch('` + server.URL + `').then(function(r) {
-		return r.text()
-	}).then(function(data) {
-		__captureSuccess(data)
-	}).catch(function (e) {
-		__captureError(e)
-	})`)
-	require.NoError(err)
-
-	select {
-	case data := <-dataCh:
-		require.True(data.IsString())
-		require.Equal(body, data.String())
-	case err := <-errCh:
-		require.Fail("request failed", err)
-	case <-time.After(1 * time.Second):
-		require.Fail("test timed out")
-	}
-}
-
-func (s *CellTestSuite) TestJailFetchCatch() {
-	require := s.Require()
-
-	cell, err := s.jail.NewCell(testChatID)
-	require.NoError(err)
-	require.NotNil(cell)
-	defer cell.Stop()
-
-	dataCh := make(chan otto.Value, 1)
-	errCh := make(chan otto.Value, 1)
-
-	err = cell.Set("__captureSuccess", func(res otto.Value) { dataCh <- res })
-	require.NoError(err)
-	err = cell.Set("__captureError", func(res otto.Value) { errCh <- res })
-	require.NoError(err)
-
-	// run JS code for fetching invalid URL
-	_, err = cell.Run(`fetch('http://👽/nonexistent').then(function(r) {
-		return r.text()
-	}).then(function(data) {
-		__captureSuccess(data)
-	}).catch(function (e) {
-		__captureError(e)
-	})`)
-	require.NoError(err)
-
-	select {
-	case data := <-dataCh:
-		require.Fail("request should have failed, but returned", data)
-	case e := <-errCh:
-		require.True(e.IsObject())
-		name, err := e.Object().Get("name")
-		require.NoError(err)
-		require.Equal("Error", name.String())
-		_, err = e.Object().Get("message")
-		require.NoError(err)
-	case <-time.After(3 * time.Second):
-		require.Fail("test timed out")
-	}
-}
-
 // TestJailFetchRace tests multiple fetch callbacks,
 // supposed to be run with '-race' flag.
-func (s *CellTestSuite) TestJailFetchRace() {
+func (s *CellTestSuite) TestCellFetchRace() {
 	body := `{"key": "value"}`
 	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Add("Content-Type", "application/json")
 		w.Write([]byte(body)) //nolint: errcheck
 	}))
 	defer server.Close()
-	require := s.Require()
-
-	cell, err := s.jail.NewCell(testChatID)
-	require.NoError(err)
-	require.NotNil(cell)
-	defer cell.Stop()
 
+	cell := s.cell
 	dataCh := make(chan otto.Value, 1)
 	errCh := make(chan otto.Value, 1)
 
-	err = cell.Set("__captureSuccess", func(res otto.Value) { dataCh <- res })
-	require.NoError(err)
+	err := cell.Set("__captureSuccess", func(res otto.Value) { dataCh <- res })
+	s.NoError(err)
```

Comment:
In order to be consistent with the whole file. What's the point in doing
```go
require := s.Require()
require.NoError(err)
```
as 
```go
s.NoError(err)
```
can be used?

Reply:
There's a reason: `s.NoError` uses the `assert` package while `s.Require().NoError` requires the `require` package. The last immediately fails the test while the first just marks the test as failing and continues execution.

Reply:
Hm, that's interesting. I like `assert` much less now. I should be more careful when I swap these two then. In this case, it does not matter.

Reply:
Sorry, what doesn't matter? I think it matters because tests with `require` allows to fail earlier.

Reply:
Generally, it does but it does not matter in this case. 

> I think it matters because tests with require allows to fail earlier.

That's true but using `assert` you can spot all errors after a single run (possibly some errors will be false positives if they depend on each other). With `require` you need to rerun tests after each change because you don't know where else there are issues. 

As our unit tests are super quick, I guess `assert` is a reasonable choice, usually.

Btw. this is a good candidate to be put into Code Styles and Code Review documents.

Reply:
Yeah, totally. You mean use `assert` for fast unit tests and `require` for long e2e tests, right?
cc @JekaMas

Reply:
Not sure if it should be that strict but it would be great to collect some use cases and tips when to use which.